### PR TITLE
Add "Join server" modal

### PR DESCRIPTION
**Pull request by Oksamies**
And the functionalities for it.
Join server opens popup
Popup has some information
Get mod man button in popup opens MM OW page

refs TS-1052

### CODE REVIEW

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (anttimaki) [inline apps/sertra/components/LaunchingModManPopup.module.css:64]: Is this needed?

File: apps/sertra/components/LaunchingModManPopup.tsx
Comment (anttimaki) [inline apps/sertra/components/LaunchingModManPopup.tsx:17]: This should probably be an icon instead of a character?

File: apps/sertra/components/LaunchingModManPopup.tsx
Comment (anttimaki) [inline apps/sertra/components/LaunchingModManPopup.tsx:31]: Is the wrapping `div` needed? At least I don't see `footerRight` defined anywhere.

File: apps/sertra/pages/server/[id].tsx
Comment (anttimaki) [inline apps/sertra/pages/server/[id].tsx:51]: I'm not sure how important it's in this particular case, but usually I'd do this as `setShowPopup((current) => !current);`, which guarantees that the toggling is done to the latest value of `showPopup` in the state and not some outdated value.

File: apps/sertra/pages/server/[id].tsx
Comment (anttimaki) [inline apps/sertra/pages/server/[id].tsx:70]: Not a big deal but I feel the name `setClose` (sets value to "closed") somewhat conflicts with `togglePopup` (sets value to "open" or "closed").

File: apps/sertra/pages/server/[id].tsx
Comment (anttimaki) [inline apps/sertra/pages/server/[id].tsx:52]: Have you disabled the modal intentionally based on the discord discussion? Just making sure this works as you think it does, since at least for me the modal doesn't currently open when the button is pressed (since the value of `showPopup` hasn't yet be updated by the line above when this line is executed. If this was intentional, then maybe it should be at least documented, but I'd prefer to comment out this whole `if` block.

**Review by anttimaki — CHANGES_REQUESTED** [review]

File: apps/sertra/components/GetModMan.tsx
Comment (Oksamies) [reply apps/sertra/components/GetModMan.tsx:3]: All of the brackets in TSX are throwing of, ty for the point out 👍

File: apps/sertra/components/GetModMan.tsx
Comment (Oksamies) [reply apps/sertra/components/GetModMan.tsx:10]: Okay so I think I've "fixed" it now. Let me know if it could have been done better. Also I'll be waiting for an answer from @gkurck for the `<a>` element thing.

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (Oksamies) [reply apps/sertra/components/LaunchingModManPopup.module.css:6]: Whoosp 👀

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (Oksamies) [reply apps/sertra/components/LaunchingModManPopup.module.css:22]: I rounded those down a bit. I just added all the decimals to make sure I got it right style wise. Though this might be something I need to mention to @gkurck , so that he can make sure none of the dimensions are non by 8 divisible.

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (Oksamies) [reply apps/sertra/components/LaunchingModManPopup.module.css:58]: Huh, it's not. Weird cause when I first add those, it wouldn't apply it to the whole of the footer. Oh well removing those dimensions. I guess I had something back then that fucked it up. Baby steps

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (Oksamies) [reply apps/sertra/components/LaunchingModManPopup.module.css:64]: Guess not, thought it would need it for the flex-grow 🤷

File: apps/sertra/components/LaunchingModManPopup.tsx
Comment (Oksamies) [reply apps/sertra/components/LaunchingModManPopup.tsx:17]: Yes true, I'll figure out how it should be done.

File: apps/sertra/components/LaunchingModManPopup.tsx
Comment (Oksamies) [reply apps/sertra/components/LaunchingModManPopup.tsx:31]: Nope, remnant, removing it 👍

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (anttimaki) [reply apps/sertra/components/LaunchingModManPopup.module.css:22]: Why do you need gkurck for that, don't you have a calculator 😅 And why would the margin need to be divisible by 8? Is that some sort of stated design goal?

File: apps/sertra/components/GetModMan.tsx
Comment (gkurck) [reply apps/sertra/components/GetModMan.tsx]: Yes, this should be `<a>`, as that's what it is - an external link. When you're dealing with either hyperlinks or external links, use `<a>`. When you are dealing with something toggle-able and/or programmable, such as submitting a form or toggling a menu, use `<button>`.

The "Join Server" -button should be `<a>` as well.

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (gkurck) [reply apps/sertra/components/LaunchingModManPopup.module.css]: The "divisible by 8" thing comes from the fact that the since the browser base pixel value is (in most cases) 16px, the measurements in the designs follow that pattern. so 1rem = 16px, .5rem = 8px and so on. This also helps to make sure the units are always divisible - compare that to base of 10, where a third would be 3.333333, which isn't ideal.

That being said, the amount of decimals here are an overkill, and you should be able to survive with divisions of 1.

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (gkurck) [reply apps/sertra/components/LaunchingModManPopup.module.css]: Also might want to consider not using the font but the `<svg>` components instead.

File: apps/sertra/components/LaunchingModManPopup.module.css
Comment (anttimaki) [reply apps/sertra/components/LaunchingModManPopup.module.css]: So yeah, it's not 8 but rather 16. And "divisible by 16" is incorrect too since 8 is not divisible by 16, but the other way around. (Not 100% sure but I'd think that's how the semantics go.)

**Review by VilppeRiskidev — APPROVED** [review]

File: apps/sertra/README.md
Comment (anttimaki) [inline apps/sertra/README.md:35]: This was left over from the original wrong placement.

File: apps/sertra/pages/server/[id].tsx
Comment (anttimaki) [inline apps/sertra/pages/server/[id].tsx:56]: Now that there's just the one command in the function this could as well be `const togglePopup = () => setShowPopup((current) => !current);`

**Review by anttimaki — APPROVED** [review]
Good to merge as-is, or if you decide to act on the two new comments, feel free to merge this yourself afterwards (assuming GitHub lets you).

**Pull request merged**
